server/user_account/views.py

```diff
@@ -64,21 +67,25 @@ def create(self, request, *args, **kwargs):
         }
 
         auth_user = CustomUser.objects.get(user_name=auth_content['user'])
-        # TODO: Limit account creation by role
-        if auth_user.role != 'SA':
-            pass
+        # Limit account creation by role
+        if (auth_user.role == 'SA') or (auth_user.role == 'IM' and (auth_user.organization_id == request.data.get('organization', ''))):
```

Comment:
Since we are working with permissions here, and the way we have it now is that all the permissions are defined inside get_permissions(self) method, I feel it would be better to keep handling them there:
- So for verifying that the user doing the request is an SA you already have the IsSystemAdmin.
- For verifying that the user is an IM and is trying to add someone in his own organization then you could create a helper for that i think that can be generalized since all the operations that an inventory manager will do will only be allowed in his organization. You can inspire yourself of what i did for IsSystemAdmin and IsInventoryManager.

What do you think @NicholasNagy ?

Reply:
Good catch!
I tried to implement what you described, please let me know if there are possible improvements.

Reply:
I agree @RobertBeaudenon , there were just a couple of extra small improvements that I asked @Amanda-Kolopa to do, but aside from that, this PR is looking good.